storage/kv.go

```diff
@@ -38,6 +42,17 @@ type KV interface {
 	TxnPut(txnID int64, key, value []byte) (rev int64, err error)
 	TxnDeleteRange(txnID int64, key, end []byte) (n, rev int64, err error)
 
+	// Watcher watches the events happening or happened in etcd. The whole
```

Comment:
This can replace the old KV api. And we only expose WatchableKV to outside world.